distribution/scripts/ciphertool.sh

```
@@ -0,0 +1,122 @@
+  if expr "$link" : '.*/.*' > /dev/null; then
+    PRG="$link"
+  else
+    PRG=`dirname "$PRG"`/"$link"
+  fi
+done
+
+# Get standard environment variables
+PRGDIR=`dirname "$PRG"`
+
+# Only set ICP_HOME if not already set
+[ -z "$ICP_HOME" ] && ICP_HOME=`cd "$PRGDIR/.." ; pwd`
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)